### DISCUSSION TITLE
Backend

### DISCUSSION THREAD

**Discussion by LOGESH-B** [gbowne1/codebooker#82] [General]
Hii @gbowne1 
Can you explain the Backend model db flow...models,relationships...if the backend is completed then its easy with frontend

**Comment by gbowne1** [comment]
@LOGESH-B 

ESlint is reporting an issue with `path` in the /server/index.js 

```shell
$ npx eslint --fix .

~/codebooker/src/server/index.js
  5:7  error  'path' is assigned a value but never used  no-unused-vars
```

**Reply by LOGESH-B** [reply]
In future it may needed for now it can be removed I will remove it on my next issue✌️

**Reply by gbowne1** [reply]
Usually, path is for express static file paths to tell express which directory to serve static files from, usually the /public.

**Comment by gbowne1** [comment]
@LOGESH-B

Need to add nodemon package (and add a nodemon.json config file) to the server to watch the express server at @ /server/index.js and reload if necessary.

**Comment by gbowne1** [comment]


**Reply by gbowne1** [reply]
oh and ESlint says 

```shell
Compiled with warnings.

[eslint] 
src/pages/Login/Login.jsx
  Line 29:33:  Expected '===' and instead saw '=='  eqeqeq

Search for the keywords to learn more about each warning.
To ignore, add // eslint-disable-next-line to the line before.

WARNING in [eslint] 
src/pages/Login/Login.jsx
  Line 29:33:  Expected '===' and instead saw '=='  eqeqeq

webpack compiled with 1 warning
```

so.. Login.jsx needs an extra = on line 29..`if (response.status === 200) {`  this will fix it and now the GET sorta works. tested it locally.

**Reply by gbowne1** [reply]
To start the project I always start in this order:

- [x] Start mongod the mongodb database server 
- [x] Start mongosh the mongodb shell 
- [x] run use codebooker command in mongosh
- [x] In root of projecct `npm start` the react dev server
- [x] in /src/server run node index.js to start the backend server

**Comment by gbowne1** [comment]
We also still have this:

```shell
$ npx eslint --fix .

/media/gbowne1/18656299-5992-400a-a7a4-b6ffc4b0612f/Documents/codebooker/src/server/controller/booksController.js
  14:28  error  'category' is assigned a value but never used  no-unused-vars

/media/gbowne1/18656299-5992-400a-a7a4-b6ffc4b0612f/Documents/codebooker/src/server/index.js
   5:7  error  'path' is assigned a value but never used  no-unused-vars
  17:7  error  'User' is assigned a value but never used  no-unused-vars
```

in /server/index.js

**Reply by gbowne1** [reply]
Oh and clicking on the ->] Logout button in the user menu (R) and should kill any session and perhaps display some message to the user? and redirect to /login.

From /login should not be able to manually go to the app root '/'  either, only if logged in.  it should be a protected route.

**Reply by LOGESH-B** [reply]
Resolved and will updated on next PR

**Reply by gbowne1** [reply]
ok

**Comment by gbowne1** [comment]
Does the books collection/model/schema work? it should save the data to the books collection.  I think we're still relying on the json that is in the code.  I get a wierd error from React.

**Reply by LOGESH-B** [reply]
@gbowne1 
Yeah its working fine in backend but we need to integrate with the user who added the book in the books model

**Reply by gbowne1** [reply]
I dont get any objects.  nothing in the `books` collection I have either. I tried to add a book. db.books.findOne() returns 'null'.

**Reply by LOGESH-B** [reply]
I tested with the **POSTMAN** its working fine with the available book routes

**Reply by gbowne1** [reply]
my POSTMAN is a bit funky with the API. it tries to use the cloud function. and yeah I'm using web.postman.co

**Reply by gbowne1** [reply]
I'd like to see that working..
